replica/database.cc

```diff
@@ -1739,8 +1760,11 @@ future<> database::apply_with_commitlog(column_family& cf, const mutation& m, db
         try {
             commitlog_entry_writer cew(m.schema(), fm, db::commitlog::force_sync::no);
             h = co_await cf.commitlog()->add_entry(m.schema()->id(), cew, timeout);
+        } catch (timed_out_error&) {
+            // see above (#9919)
+            throw_commitlog_add_error<wrapped_timed_out_error>(cf.schema(), fm);
         } catch (...) {
-            throw_commitlog_add_error(cf.schema(), fm);
+            throw_commitlog_add_error<>(cf.schema(), fm);
```

Comment:
I think it would be clearer to have no default for the template parameter
and pass std::runtime_error here explicitly.

Reply:
Sure.